### PR TITLE
Add `Async#executor`

### PR DESCRIPTION
**Pull request by armanbilge**
This is helpful for Java interop. See:
- https://github.com/typelevel/cats-effect/pull/2919#issuecomment-1101390355
- https://github.com/http4s/http4s-jdk-http-client/pull/641

/cc @wjoel

### CODE REVIEW

File: kernel/shared/src/main/scala/cats/effect/kernel/Async.scala [hunk 1]
```diff
@@ -139,6 +140,14 @@ trait Async[F[_]] extends AsyncPlatform[F] with Sync[F] with Temporal[F] {
    */
   def executionContext: F[ExecutionContext]
 
+  /**
+   * Obtain a reference to the current execution context as a `java.util.concurrent.Executor]]`.
```
Comment (armanbilge) [inline kernel/shared/src/main/scala/cats/effect/kernel/Async.scala]:
```suggestion
   * Obtain a reference to the current execution context as a `java.util.concurrent.Executor`.
```

**Comment by djspiewak** [comment]
Does this have to be source-split for ScalaJS?

**Comment by armanbilge** [comment]
No, interestingly enough the `MacrotaskExecutor` is already an `ExecutionContextExecutor` :)

**Comment by wjoel** [comment]
Looks great, much cleaner than my PR. :+1:

**Review by djspiewak — APPROVED** [review]

**Pull request merged**